### ISSUE TITLE
[BUG] Search Freezes

### ISSUE THREAD

**Issue by amantinband** [scambier/obsidian-omnisearch#287]
First of all, thanks for this awesome plugin 🫶

**Problem description:**

Omnisearch started freezing a few days ago. This happens often but not always:
![image](https://github.com/scambier/obsidian-omnisearch/assets/39407673/aa26a9cc-a18f-44bd-aec6-195dfa8dae73)

I've tried toggling all the various settings, uninstalling/installing - nothing worked so far.

**Your environment:**

- Omnisearch version: 1.17.0
- Obsidian version: 1.4.5
- Operating system: MacOS
- Number of indexed documents in your vault (approx.): 1000

**Things to try:**

- Does the problem occur when Omnisearch is the only active community plugin:  yep
- Does the problem occur when you _don't_ index PDFs, images, or other non-notes files:  yep
- Does the problem occur after a cache reset: yep

**Comment by amantinband** [comment]
I just noticed that it freezes whenever I try searching for the word "Curriculum". 

When using the regular search, I see it appears only in 34 files. Pasting here the files names although it's probably not relevant:
* 1.3. Who is this course for and prerequisites
* 19. Domain Driven Design's Holy Grail
* 2023-09-07
* 2023-09-10
* 2023-09-11
* Advanced DDD Course
* ASP.NET Core 6 Clean Architecture
* Big Picture
* Clean Architecture Tech Train
* Clean Architecture Tech Train Curriculum
* Curriculum
* Curriculum
* Curriculum Draft (Clean Architecture Deep Dive)
* Curriculum Draft (Clean Architecture Getting Started)
* Curriculum Playground
* Getting Started With DDD Course
* Getting Started with DDD promo text
* Notion scratchpad
* Talking points

Going through the files, I don't see anything special. The only 2 files that may be suspicious are the "database folder" files (note that omnisearch freezes also when the database folder plugin is disabled).
Here is one of the files for example:
---

database-plugin: basic

---

```yaml:dbfolder
name: new database
description: new description
columns:
  __file__:
    key: __file__
    id: __file__
    input: markdown
    label: File
    accessorKey: __file__
    isMetadata: true
    skipPersist: false
    isDragDisabled: false
    csvCandidate: true
    position: 2
    isHidden: false
    sortIndex: -1
    config:
      enable_media_view: true
      link_alias_enabled: true
      media_width: 100
      media_height: 100
      isInline: true
      task_hide_completed: true
      footer_type: none
      persist_changes: false
  SectionTitle:
    input: text
    accessorKey: SectionTitle
    key: SectionTitle
    id: SectionTitle
    label: Section Title
    position: 1
    skipPersist: false
    isHidden: false
    sortIndex: -1
    width: 153
    config:
      enable_media_view: true
      link_alias_enabled: true
      media_width: 100
      media_height: 100
      isInline: false
      task_hide_completed: true
      footer_type: none
      persist_changes: false
      wrap_content: true
  Status:
    input: select
    accessorKey: Status
    key: Status
    id: Status
    label: Status
    position: 100
    skipPersist: false
    isHidden: false
    sortIndex: -1
    width: 182
    options:
      - { label: "SlidesReady", value: "SlidesReady", color: "hsl(207, 95%, 90%)"}
      - { label: "Filmed", value: "Filmed", color: "hsl(162, 95%, 90%)"}
      - { label: "Edited", value: "Edited", color: "hsl(301, 95%, 90%)"}
    config:
      enable_media_view: true
      link_alias_enabled: true
      media_width: 100
      media_height: 100
      isInline: false
      task_hide_completed: true
      footer_type: none
      persist_changes: false
      option_source: manual
config:
  remove_field_when_delete_column: false
  cell_size: normal
  sticky_first_column: false
  group_folder_column: 
  remove_empty_folders: false
  automatically_group_files: false
  hoist_files_with_empty_attributes: true
  show_metadata_created: false
  show_metadata_modified: false
  show_metadata_tasks: false
  show_metadata_inlinks: false
  show_metadata_outlinks: false
  show_metadata_tags: false
  source_data: current_folder
  source_form_result: 
  source_destination_path: /
  row_templates_folder: /
  current_row_template: 
  pagination_size: 200
  font_size: 16
  enable_js_formulas: false
  formula_folder_path: /
  inline_default: false
  inline_new_position: last_field
  date_format: yyyy-MM-dd
  datetime_format: "yyyy-MM-dd HH:mm:ss"
  metadata_date_format: "yyyy-MM-dd HH:mm:ss"
  enable_footer: false
  implementation: default
filters:
  enabled: false
  conditions:
```

**Comment by scambier** [comment]
Hello, can you enable "verbose logging" in the omnisearch options, display the developer console with ctrl+shift+i, and show me what happens when you search for "curriculum"? Thank you :)

**Comment by amantinband** [comment]
Yeah was gonna and forgot 🤦 here you go:
![image](https://github.com/scambier/obsidian-omnisearch/assets/39407673/f4803538-0ab3-47c2-a920-1a3f96b2a340)

```
Uncaught (in promise) SyntaxError: Invalid regular expression: /\bcurrentcount++\b/: Nothing to repeat (at plugin:omnisearch:39:4693)
    at new RegExp (<anonymous>)
    at eval (plugin:omnisearch:39:4693)
    at Array.forEach (<anonymous>)
    at kt (plugin:omnisearch:39:4675)
    at af (plugin:omnisearch:42:28168)
    at Array.Uv (plugin:omnisearch:42:28769)
    at pr (plugin:omnisearch:7:4852)
    at Vp (plugin:omnisearch:42:2420)
    at Oe (plugin:omnisearch:7:8809)
    at new Sl (plugin:omnisearch:42:3510)
e
```

**Comment by scambier** [comment]
This should be fixed in 1.17.1 (and thanks for your sponsoring 😊)

**Comment by amantinband** [comment]
Checked now and it works. Thanks, Simon!
